src/Symfony/Component/Intl/Util/GitRepository.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Intl\Util;
+
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\Intl\Exception\RuntimeException;
+
+/**
+ * @internal
+ */
+final class GitRepository
+{
+    private $path;
+
+    /**
+     * @param string $path
+     */
+    public function __construct($path)
+    {
+        $this->path = $path;
+
+        $this->getUrl();
+    }
+
+    /**
+     * @param string $remote
+     * @param string $targetDir
+     *
+     * @return GitRepository
+     */
+    public static function download($remote, $targetDir)
+    {
+        static::exec('which git', 'The command "git" is not installed.');
```

Comment:
Must be `self::exec` as it accesses a private property. A `static::`  access to a private method would break when subclassing and so should always be avoided.
The fact that the class is final means that the buggy case cannot be reached, but that's still a bad idea to show such code in the wild IMO (plus, it moves the resolution to runtime when it could be done at compile time)